app/components/TrackCard/tests/index.test.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { fireEvent } from '@testing-library/dom';
+import { renderProvider, timeout } from '@app/utils/testUtils';
+import TrackCard from '../index';
+
+const trackDetails = {
+  trackName: 'Ignite',
+  artistName: 'Alan Walker',
+  artworkUrl100: 'some url string',
+  previewUrl: 'another url string'
+};
+
+describe('Tests for TrackCard component', () => {
+  it('should render the component with all the necessary elements', () => {
+    const { getByRole, getByText } = renderProvider(<TrackCard trackDetails={trackDetails} />);
+    const artistRegex = new RegExp('Alan Walker', 'i');
+
+    const trackName = getByText('Ignite');
+    const artistName = getByText(artistRegex);
+    const trackImage = getByRole('img');
+    const previewBtn = getByRole('button', {
+      value: { text: 'Preview' }
+    });
+    const audioElement = getByRole('audio');
+
+    expect(trackImage).toBeInTheDocument();
+    expect(trackName).toHaveTextContent(trackDetails.trackName);
+    expect(artistName).toHaveTextContent(trackDetails.artistName);
+    expect(previewBtn).toBeInTheDocument();
+    expect(audioElement).toBeInTheDocument();
+  });
+
+  it('should display correct text in the preview/pause button', async () => {
+    let previewBtn;
+
+    // Initial text should be 'Preview'
+    const { getByRole } = renderProvider(<TrackCard trackDetails={trackDetails} />);
+    previewBtn = getByRole('button', {
+      value: { text: 'Preview' }
+    });
+    expect(previewBtn).toHaveTextContent('Preview');
+
+    // Text should change to 'Pause' if button is clicked
+    fireEvent.click(previewBtn);
+    await timeout(100);
+    previewBtn = getByRole('button', {
+      value: { text: 'Pause' }
+    });
+    expect(previewBtn).toHaveTextContent('Pause');
```

Comment:
We are already getting the button by text `Pause`. It doesn't make sense to check if the button has the text content `Pause`. Instead of doing this you can just check if a button with text `Pause` exists in the document.
Please update this in other places as well.